diff --git a/arch/arm/boot/dts/overlays/Makefile b/arch/arm/boot/dts/overlays/Makefile
index c890e5d..43288d3 100644
--- a/arch/arm/boot/dts/overlays/Makefile
+++ b/arch/arm/boot/dts/overlays/Makefile
@@ -53,6 +53,7 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	mcp3008.dtbo \
 	midi-uart0.dtbo \
 	mmc.dtbo \
+	monome.dtbo \
 	mz61581.dtbo \
 	pi3-act-led.dtbo \
 	pi3-disable-bt.dtbo \
diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index 970c9c9..726046f 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -812,6 +812,12 @@ Params: overclock_50            Clock (in MHz) to use when the MMC framework
                                 requests 50MHz
 
 
+Name:   monome
+Info:   Configures the Monome audio card
+Load:   dtoverlay=monome
+Params: <None>
+
+
 Name:   mz61581
 Info:   MZ61581 display by Tontec
 Load:   dtoverlay=mz61581,<param>=<val>
diff --git a/sound/soc/bcm/Kconfig b/sound/soc/bcm/Kconfig
index e626da6..e7e0911 100644
--- a/sound/soc/bcm/Kconfig
+++ b/sound/soc/bcm/Kconfig
@@ -154,6 +154,13 @@ config SND_BCM2708_SOC_FE_PI_AUDIO
 	help
 	  Say Y or M if you want to add support for Fe-Pi-Audio.
 
+config SND_BCM2708_SOC_MONOME
+       tristate "Support for monome-snd"
+       depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
+       select SND_SOC_CS4270
+       help
+         Say Y or M if you want to add support for Monome soundcard
+
 config SND_PISOUND
 	tristate "Support for Blokas Labs pisound"
 	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
diff --git a/sound/soc/bcm/Makefile b/sound/soc/bcm/Makefile
index 8d2d207..6bd6003 100644
--- a/sound/soc/bcm/Makefile
+++ b/sound/soc/bcm/Makefile
@@ -29,6 +29,7 @@ snd-soc-dionaudio-loco-v2-objs := dionaudio_loco-v2.o
 snd-soc-allo-piano-dac-objs := allo-piano-dac.o
 snd-soc-pisound-objs := pisound.o
 snd-soc-fe-pi-audio-objs := fe-pi-audio.o
+snd-soc-monome-objs := monome.o
 
 obj-$(CONFIG_SND_BCM2708_SOC_ADAU1977_ADC) += snd-soc-adau1977-adc.o
 obj-$(CONFIG_SND_BCM2708_SOC_HIFIBERRY_AMP) += snd-soc-hifiberry-amp.o
@@ -50,3 +51,4 @@ obj-$(CONFIG_SND_BCM2708_SOC_DIONAUDIO_LOCO_V2) += snd-soc-dionaudio-loco-v2.o
 obj-$(CONFIG_SND_BCM2708_SOC_ALLO_PIANO_DAC) += snd-soc-allo-piano-dac.o
 obj-$(CONFIG_SND_PISOUND) += snd-soc-pisound.o
 obj-$(CONFIG_SND_BCM2708_SOC_FE_PI_AUDIO) += snd-soc-fe-pi-audio.o
+obj-$(CONFIG_SND_BCM2708_SOC_MONOME) += snd-soc-monome.o
